src/Sylius/Behat/Context/Api/Admin/ManagingTaxonImagesContext.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\RequestBuilder;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Context\Api\Resources;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Request;
+use Webmozart\Assert\Assert;
+
+final class ManagingTaxonImagesContext implements Context
+{
+    public function __construct(
+        private ApiClientInterface $client,
+        private ResponseCheckerInterface $responseChecker,
+        private SharedStorageInterface $sharedStorage,
+        private \ArrayAccess $minkParameters,
+    ) {
+    }
+
+    /**
+     * @When /^I attach the "([^"]+)" image with "([^"]+)" type to (this taxon)$/
+     */
+    public function iAttachTheImageWithTypeToThisTaxon(string $path, ?string $type, TaxonInterface $taxon): void
+    {
+        $builder = RequestBuilder::create(
+            sprintf('/api/v2/admin/taxons/%s/images', $taxon->getCode()),
+            Request::METHOD_POST,
+        );
+        $builder->withHeader('CONTENT_TYPE', 'multipart/form-data');
+        $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+        $builder->withHeader('HTTP_Authorization', 'Bearer ' . $this->sharedStorage->get('token'));
+        $builder->withFile('file', new UploadedFile($this->minkParameters['files_path'] . $path, basename($path)));
+
+        if (null !== $type) {
+            $builder->withParameter('type', $type);
+        }
+
+        $this->client->request($builder->build());
```

Comment:
What about using a `sharedStorage` to keep the prepared request and then use it in the generic `SaveContext`
Something like that:
```php
    public function iSaveMyChanges(): void
    {
        if ($this->sharedStorage->has('request')) {
            $this->client->request($this->sharedStorage->get('request'));

            return;
        }

        $this->client->update();
    }
```
Thanks to this, we no longer need an empty save step for the API

Reply:
Not in every scenario would this solve the problem, as we sometimes do several requests with uploads, so this would create a new problem and would not be a clearer solution at all IMO